src/runtime/virtcontainers/clh.go

```diff
@@ -476,12 +476,21 @@ func (clh *cloudHypervisor) CreateVM(ctx context.Context, id string, network Net
 	// Create the VM config via the constructor to ensure default values are properly assigned
 	clh.vmconfig = *chclient.NewVmConfig(*chclient.NewPayloadConfig())
 
-	// Make sure the kernel path is valid
-	kernelPath, err := clh.config.KernelAssetPath()
+	// Make sure the igvm path is valid
```

Comment:
I think exactly here we need to either make a decision based on the confidential_guest field, or based on whether IGVM is specified. At this current state we can't usptream it as it breaks vanilla kata

Reply:
Why does this break vanilla kata? The kernel path check is now moved inside of this IGVM conditional. So if no IGVM is set it goes on to check that the kernel is set as usual